geoportailv3/static/js/layerpermalinkservice.js

```diff
@@ -0,0 +1,254 @@
+    return layer.getOpacity();
+  });
+  var bgLayer = this.backgroundLayerMgr_.get(this.map_);
+  this.stateManager_.updateState({
+    'layers': layerIds.join('-'),
+    'opacities': opacities.join('-'),
+    'bgLayer': bgLayer.get('label')
+  });
+};
+
+
+/**
+ * @param {Array.<number>} layerIds
+ * @param {Array.<Object>} themes
+ * @private
+ */
+app.LayerPermalinkManager.prototype.applyLayerStateToMap_ =
+    function(layerIds, themes) {
+  var flatCatalogue = [];
+  for (var i = 0; i < themes.length; i++) {
+    var theme = themes[i];
+    goog.array.extend(flatCatalogue,
+        app.LayerPermalinkManager.getAllChildren_(theme.children)
+    );
+  }
+  goog.array.forEach(layerIds.reverse(),
+      function(layerId, layerIndex) {
+        var node = goog.array.find(flatCatalogue, function(catItem) {
+          return catItem.id === layerId;
+        });
+        var layer = this.getLayerFunc_(node);
+        // set opacity trough metadata to not interfere
+        // with the layer opacity manager service
+        layer.get('metadata')['start_opacity'] =
+            this.getStateValue_('opacities').reverse()[layerIndex];
+        this.map_.addLayer(layer);
+      }, this
+  );
+};
+
+
+/**
+ * @param {string} parameter
+ * @private
+ * @return {Array.<number>|undefined}
+ */
+app.LayerPermalinkManager.prototype.getStateValue_ = function(parameter) {
+  var result = '';
+  var response = this.stateManager_.getInitialValue(parameter);
+  if (goog.isDef(response) && response.length > 0) {
+    result = this.stateManager_.getInitialValue(parameter);
+  } else {
+    return undefined;
+  }
+  var items = [];
+  goog.array.forEach(result.split('-'), function(string) {
+    var value = parseFloat(string);
+    if (goog.isNumber(value)) {
+      items.push(value);
+    }
+  });
+  if (items.length === 0) {
+    return undefined;
+  } else {
+    return items;
+  }
+};
+
+
+/**
+ * @param {Array.<ol.layer.Layer>} selectedLayers
+ * @private
+ */
+app.LayerPermalinkManager.prototype.setupWatchers_ = function(selectedLayers) {
+
+  goog.events.listen(this.backgroundLayerMgr_, ngeo.BackgroundEventType.CHANGE,
+      function() {
+        var bgLayer = this.backgroundLayerMgr_.get(this.map_);
+        this.stateManager_.updateState({
+          'bgLayer': bgLayer.get('label')
+        });
+      }, undefined, this);
+
+  this.scope_.$watchCollection(goog.bind(function() {
+    return selectedLayers;
+  }, this), goog.bind(function() {
+    this.setLayerState_(selectedLayers);
+  }, this));
+
+  var layers = this.map_.getLayers();
+  // Add event listeners to existing selected layers
+  goog.array.forEach(layers.getArray(), function(layer) {
+    goog.events.listen(layer, ol.ObjectEventType.PROPERTYCHANGE,
+        goog.bind(function() {
+          this.setLayerState_(selectedLayers);
+        }, this));
+  }, this);
+  // Add event listener to all future selected layers
+  goog.events.listen(layers, ol.CollectionEventType.ADD,
+      goog.bind(
```

Comment:
You don't need to use `goog.bind` here as you pass `this` to `goog.events.listen` as the fifth argument (as done elsewhere).